src/sqlfluff/core/templaters/placeholder.py

```diff
@@ -153,14 +153,7 @@ def process(
             else:
                 param_name = found_param["param_name"]
             last_literal_length = span[0] - last_pos_raw
-            try:
-                replacement = str(context[param_name])
-            except KeyError as err:
-                # TODO: Add a url here so people can get more help.
-                raise SQLTemplaterError(
-                    "Failure in placeholder templating: {}. Have you configured your "
-                    "variables?".format(err)
-                )
+            replacement = str(context.get(param_name, ""))
```

Comment:
It might be better to set custom default value instead of blank string, but I'd like to hear your opinion.

Reply:
@shyaginuma - I think it would be safer to replace with the name of the variable as a string. That way it's transparent and most likely to still parse. This is similar to what we've done with the dbt `ref()` macro.

i.e. `SELECT :foo, 1` would get parsed as `SELECT "foo", 1` rather than `SELECT , 1` (the latter would fail to parse, but the earlier one would pass).

How does that sound to you? I think most SQL dialects allow quoted literals in most cases that would accept strings, dates or other datatypes.

@barrywhart - do you have any views as you put up one of the original issues?

Reply:
IIRC, when I implemented the similar behavior for the Jinja and dbt templaters, I had it simply return the unquoted string. Either behavior may be reasonable, but I think we should be consistent. Can someone confirm the dbt templater's behavior?

Reply:
@alanmcruickshank Thanks for your suggestion! Fixed in [`31bd5f1`](https://github.com/sqlfluff/sqlfluff/pull/5101/commits/31bd5f12b063a999c0eba0ecbdd296a109fc7ab5), could you take a look again?